src/components/forms/verify-two-fa-code-form.tsx

```diff
@@ -1,31 +1,80 @@
+// import { useRouter } from "next/router";
 import React from "react";
 import styled from "styled-components";
 import { v4 as uuidv4 } from "uuid";
 
 import { InputText } from "../input/input-text";
+import { WrongInputError } from "../input/wrong-input-error";
 import { Form } from "./form";
+import { HttpVerbs } from "@src/@types/http-verbs";
 import { Button, ButtonVariant } from "@src/components/buttons/buttons";
+import { deviceBreakpoints } from "@src/design-system/theme";
+import { ERRORS_DATA } from "@src/errors/web-errors";
+import { fetchJson } from "@src/utils/fetch-json";
 
-const VerifyTwoFACodeForm: React.FC = () => {
+const VerifyTwoFACodeForm: React.FC<{
+  setIsCodeSent: React.Dispatch<React.SetStateAction<boolean>>;
+}> = ({ setIsCodeSent }) => {
   const [formID, setFormID] = React.useState<string>(uuidv4());
   const [verificationCode, setVerificationCode] = React.useState<string>("");
+  const [errorMessage, setErrorMessage] = React.useState<string | null>(null);
+
+  // const router = useRouter();
 
   return (
     <ExtendedStyledForm
       formID={formID}
       onSubmit={async () => {
         setFormID(uuidv4());
-        return;
+
+        await fetchJson(
+          "/api/auth-connect/verify-two-fa-validation-code",
+          HttpVerbs.POST,
+          { verificationCode },
+        )
+          .then((response) => response.json())
+          .then((parsedResponse) => {
+            if ("message" in parsedResponse) {
+              if (
+                parsedResponse.message === ERRORS_DATA.INVALID_BODY.message ||
+                parsedResponse.message ===
+                  ERRORS_DATA.INVALID_VALIDATION_CODE.message
+              ) {
+                setFormID(uuidv4());
+                setErrorMessage("Invalid validation code");
+              }
+
+              if (
+                parsedResponse.message ===
+                ERRORS_DATA.SUDO_EVENT_IDS_NOT_FOUND.message
+              ) {
+                setFormID(uuidv4());
+                setIsCodeSent(false);
+              }
+            }
+
+            // router && router.push("/account/security/update");
+          });
```

Comment:
What about using this here https://github.com/fewlinesco/connect-account/blob/d79e4d33d1eabd576ac6a239c25212d0521a7cdd/src/components/forms/validate-identity-form.tsx
To uniform the way we do it. What do you think?

Reply:
Absolutely.